### ISSUE TITLE
Not able to attach volume of filesystem ext4dev to a container

### ISSUE THREAD

**Issue by ashahi1** [vmware/docker-volume-vsphere#1617]
Steps:
1. Created a volume of file system type ext4dev
2. Tried attaching volume to a container - error: ```Error response from daemon: error while mounting volume '/mnt/vmdk/volume1@sharedVmfs-0': VolumeDriver.Mount: Failed to mount device /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0 at /mnt/vmdk/volume1@sharedVmfs-0: no such device.```

-------------------

Steps and their output are as follows:

1. Create a volume of file system ext4dev
```
root@sc-rdops-vm02-dhcp-52-237:~# docker volume create --driver=vsphere --name=volume1 -o size=1gb -o fstype=ext4dev
volume1
root@sc-rdops-vm02-dhcp-52-237:~#
root@sc-rdops-vm02-dhcp-52-237:~# docker volume ls
DRIVER              VOLUME NAME
vsphere:latest      volume1@sharedVmfs-0
root@sc-rdops-vm02-dhcp-52-237:~#
root@sc-rdops-vm02-dhcp-52-237:~# docker volume inspect volume1@sharedVmfs-0
[
    {
        "Driver": "vsphere:latest",
        "Labels": null,
        "Mountpoint": "/mnt/vmdk/volume1@sharedVmfs-0",
        "Name": "volume1@sharedVmfs-0",
        "Options": {},
        "Scope": "global",
        "Status": {
            "access": "read-write",
            "attach-as": "independent_persistent",
            "capacity": {
                "allocated": "39MB",
                "size": "1GB"
            },
            "clone-from": "None",
            "created": "Fri Jul 21 06:58:05 2017",
            "created by VM": "ubuntu-VM0.0",
            "datastore": "sharedVmfs-0",
            "diskformat": "thin",
            "fstype": "ext4dev",     <<<<<<<<<<<<<<<<<<<<
            "status": "detached"
        }
    }
]
```

2. Tried to run a container with volume attached - failed
```
root@sc-rdops-vm02-dhcp-52-237:~# docker run -it --volume-driver=vsphere -v volume1@sharedVmfs-0:/vol1 --name ub_FS ubuntu
docker: Error response from daemon: error while mounting volume '/mnt/vmdk/volume1@sharedVmfs-0': VolumeDriver.Mount: Failed to mount device /dev/disk/by-path/pci-0000:03:00.0-scsi-0:0:0:0 at /mnt/vmdk/volume1@sharedVmfs-0: no such device.
```

---------------------

Logs:
[docker-volume-vsphere.txt](https://github.com/vmware/docker-volume-vsphere/files/1164715/docker-volume-vsphere.txt)
[vmdk_ops.txt](https://github.com/vmware/docker-volume-vsphere/files/1164716/vmdk_ops.txt)

**Comment by govint** [comment]
@ashahi1 can you post the logs.

**Comment by ashahi1** [comment]
@govint That was quick. I was in the middle of updating logs. :)

**Comment by govint** [comment]
Seems _ext4dev_ needs to be mounted as _ext4_,

root@photon-machine [ /vol/vpl-testbed/docker-volume-vsphere ]# mount /dev/sdc /mnt
mount: unknown filesystem type 'ext4dev'
root@photon-machine [ /vol/vpl-testbed/docker-volume-vsphere ]# mount -t ext4 /dev/sdc /mnt
root@photon-machine [ /vol/vpl-testbed/docker-volume-vsphere ]#

Do we need a code fix for this? I'd say no. Document it maybe.

**Comment by govint** [comment]
ext4dev can be converted to ext4 via fsck,

root@photon-machine [ /vol/vpl-testbed/docker-volume-vsphere ]# lsblk -f /dev/sdc
NAME FSTYPE  LABEL UUID                                 MOUNTPOINT
sdc  ext4dev       a2745b1f-f8d5-4804-a7a3-9edb0c569f9e <----------ext4dev

root@photon-machine [ /vol/vpl-testbed/docker-volume-vsphere ]# fsck /dev/sdc
fsck from util-linux 2.27.1
e2fsck 1.42.13 (17-May-2015)
The test_fs flag is set (and ext4 is available).  Clear<y>? yes
Pass 1: Checking inodes, blocks, and sizes
Pass 2: Checking directory structure
Pass 3: Checking directory connectivity
Pass 4: Checking reference counts
Pass 5: Checking group summary information
/dev/sdc: 11/65536 files (0.0% non-contiguous), 12635/262144 blocks

root@photon-machine [ /vol/vpl-testbed/docker-volume-vsphere ]# lsblk -f /dev/sdc
NAME FSTYPE LABEL UUID                                 MOUNTPOINT
sdc  ext4         a2745b1f-f8d5-4804-a7a3-9edb0c569f9e <------its ext4 now

**Comment by ashahi1** [comment]
@govint Can you please tell if we are going to fix this or we need to document this?

**Comment by govint** [comment]
Suggest no code change and only documentation.

**Comment by ashahi1** [comment]
Fixed as part of PR #1689
